store/sqlstore/compliance_store.go

```diff
@@ -211,3 +211,109 @@ func (s SqlComplianceStore) ComplianceExport(job *model.Compliance) store.StoreC
 		}
 	})
 }
+
+func (s SqlComplianceStore) MessageExport(after int64, limit int64) store.StoreChannel {
+	storeChannel := make(store.StoreChannel, 1)
+
+	go func() {
+		props := map[string]interface{}{"StartTime": after, "Limit": limit}
+
+		queryPrefix :=
+			`SELECT
+				Posts.Id AS PostId,
+				Posts.CreateAt AS PostCreateAt,
+				Posts.UpdateAt AS PostUpdateAt,
+				Posts.DeleteAt AS PostDeleteAt,
+				Posts.RootId AS PostRootId,
+				Posts.ParentId AS PostParentId,
+				Posts.OriginalId AS PostOriginalId,
+				Posts.Message AS PostMessage,
+				Posts.Type AS PostType,
+				Posts.Props AS PostProps,
+				Posts.Hashtags AS PostHashtags,
+				Posts.FileIds AS PostFileIds,
+				Channels.Id AS ChannelId,
+				Channels.CreateAt AS ChannelCreateAt,
+				Channels.UpdateAt AS ChannelUpdateAt,
+				Channels.DeleteAt AS ChannelDeleteAt,
+				Channels.DisplayName AS ChannelDisplayName,
+				Channels.Name AS ChannelName,
+				Channels.Header AS ChannelHeader,
+				Channels.Purpose AS ChannelPurpose,
+				Channels.LastPostAt AS ChannelLastPostAt,
+				Users.Id AS UserId,
+				Users.CreateAt AS UserCreateAt,
+				Users.UpdateAt AS UserUpdateAt,
+				Users.DeleteAt AS UserDeleteAt,
+				Users.Username AS UserUsername,
+				Users.Email AS UserEmail,
+				Users.Nickname AS UserNickname,
+				Users.FirstName AS UserFirstName,
+				Users.LastName AS UserLastName,
+				Teams.Id AS TeamId,
+				Teams.CreateAt AS TeamCreateAt,
+				Teams.UpdateAt AS TeamUpdateAt,
+				Teams.DeleteAt AS TeamDeleteAt,
+				Teams.DisplayName AS TeamDisplayName,
+				Teams.Name AS TeamName,
+				Teams.Description AS TeamDescription,
+				Teams.AllowOpenInvite AS TeamAllowOpenInvite, `
+
+		queryMySql :=
+			`CASE
+				WHEN Posts.Type = 'system_add_to_channel'
+				THEN (SELECT Email FROM Users WHERE Username = JSON_UNQUOTE(JSON_EXTRACT(CAST(Posts.Props AS JSON), '$.addedUsername')))
+				ELSE NULL END AS AddedUserEmail,
+			CASE
+				WHEN Posts.Type = 'system_remove_from_channel'
+				THEN (SELECT Email FROM Users WHERE Username = JSON_UNQUOTE(JSON_EXTRACT(CAST(Posts.Props AS JSON), '$.removedUsername')))
+				ELSE NULL END AS RemovedUserEmail `
+
+		queryPostgres :=
+			`CASE
+				WHEN Posts.Type = 'system_add_to_channel'
+				THEN (SELECT Email FROM Users WHERE Username = JSON_EXTRACT_PATH_TEXT(CAST(Posts.Props AS json), 'addedUsername'))
+				ELSE NULL END AS AddedUserEmail,
+			CASE
+				WHEN Posts.Type = 'system_remove_from_channel'
+				THEN (SELECT Email FROM Users WHERE Username = JSON_EXTRACT_PATH_TEXT(CAST(Posts.Props AS json), 'removedUsername'))
+				ELSE NULL END AS RemovedUserEmail `
+
+		querySuffix :=
+			`FROM
+				Posts
+				LEFT OUTER JOIN Channels ON Posts.ChannelId = Channels.Id
+				LEFT OUTER JOIN Users ON Posts.UserId = Users.Id
+				LEFT OUTER JOIN Teams ON Channels.TeamId = Teams.Id
+			WHERE
+				Posts.CreateAt > :StartTime
+			ORDER BY PostCreateAt
+			LIMIT :Limit`
```

Comment:
We're going to need to loadtest this :)

Not blocking for this review though - we can merge first and iterate on it before the feature comes out of "beta" as we did with Elasticsearch and are doing currently with Data Retention.

Reply:
Yup, @coreyhulen has been very clear that this feature is to remain in beta until we know that it can work at the scale that some of our larger customers require. I've already started talking to @crspeller about how to conduct the necessary load tests.

Reply:
Ya, something tells me this is going to have to be worked on. Let's hold off merging until we have tested this.

Reply:
Oh hey guys, what's goin' on h–

![Mother of god...](http://www.reactiongifs.com/r/mog1.gif)

That query...